platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/annotations/Bubble.java

```diff
@@ -0,0 +1,328 @@
+package com.mapbox.mapboxsdk.annotations;
+
+import android.graphics.Canvas;
+import android.graphics.ColorFilter;
+import android.graphics.Paint;
+import android.graphics.Path;
+import android.graphics.PixelFormat;
+import android.graphics.Rect;
+import android.graphics.RectF;
+import android.graphics.drawable.Drawable;
+
+class Bubble extends Drawable {
+
+  private RectF mRect;
+  private Path mPath = new Path();
+  private Paint mPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
+  private Path mStrokePath;
+  private Paint mStrokePaint;
+  private float mArrowWidth;
+  private float mCornersRadius;
+  private float mArrowHeight;
+  private float mArrowPosition;
+  private float mStrokeWidth;
```

Comment:
Prefixing field names with `m` is an AOSP code convention and this project uses standard java field naming conventions. More information why we aren't using AOSP conventions anymore in this [blog](http://jakewharton.com/just-say-no-to-hungarian-notation/) from Jake Wharton.